src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -48,6 +48,56 @@ static class BitOperations
             19, 27, 23, 06, 26, 05, 04, 31
         };
 
+        /// <summary>
+        /// Evaluate whether a given integral value is a power of 2.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsPow2(int value) => (value & (value - 1)) == 0 && value >= 0;
+
+        /// <summary>
+        /// Evaluate whether a given integral value is a power of 2.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CLSCompliant(false)]
+        public static bool IsPow2(uint value)
+        {
+            if (Popcnt.IsSupported)
```

Comment:
PopCount also has a path that special-cases AdvSimd.Arm64.IsSupported.  Is that not worth checking for here?

Reply:
My current stance is that there looks to be some open questions around `popcnt` perf, particularly with regards to random inputs: https://github.com/dotnet/runtime/pull/36163#issuecomment-763813901, and so it might be better to just not have `Popcnt` until we can get some more information as to what's causing this.

That is, is it due to `popcnt` or is it due to bad branch prediction?
If it's the latter, then what is the "optimal" ordering based on expected inputs?

I think that these can largely be answered in a follow PR given the perf difference, even in the winning cases, is fairly minor.

When we do get to looking at that, I'd expect that IsPow2 is largely used with validation and with positive inputs for the `int` case, in which case doing the "popcnt" check before the "edge case check" would be better.
This is because the "popcnt" check will cover everything except `int.MinValue` (`0x8000_0000`).
This means that if the assumption about inputs is correct, then the "expected" case will be you always do two compares and in the case where it's wrong, the first check results in an early exit in more cases than the "edge case check".

Reply:
Note that Aarch64 has no `POPCNT` instruction for general-purpose registers. It would normally take 4 instructions: `fmov`, `cnt`, `addv`, `fmov`, where the first `fmov` instruction copies the value from a general-purpose register to a SIMD register and the last `fmov` instruction copies the calculated population count back to a general-purpose register for a comparison.

@tannergooding Should you run tests again, please try the `IsPow2B` variation I mentioned above.